libs/cosmos-sdk/baseapp/baseapp_parallel.go

```diff
@@ -504,3 +525,34 @@ func (l *LogForParallel) PrintLog() {
 	fmt.Println("BestBlock", l.bestBlock.string(), "Concurrency Rate", 1-float64(l.bestBlock.reRunTxs)/float64(l.bestBlock.txs))
 	fmt.Println("TerribleBlock", l.terribleBlock.string(), "Concurrency Rate", 1-float64(l.terribleBlock.reRunTxs)/float64(l.terribleBlock.txs))
 }
+
+type blockSender struct {
+	mu            sync.RWMutex
+	blockTxSender map[string]sdk.SigCache
+}
+
+func newBlockSender() *blockSender {
+	return &blockSender{
+		mu:            sync.RWMutex{},
+		blockTxSender: make(map[string]sdk.SigCache),
+	}
+}
+func (b *blockSender) clear() {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	for key := range b.blockTxSender {
+		delete(b.blockTxSender, key)
+	}
```

Comment:
it`s better to make a new map here?

Reply:
https://github.com/golang/go/issues/20138